funnel/views/project.py

```diff
@@ -62,134 +62,139 @@ def project_new(profile):
+        proposals = Proposal.query.filter_by(project=self.obj).order_by(db.desc('created_at')).all()
+        return jsonp(**{
+            'project': project_data(self.obj),
+            'space': project_data(self.obj),  # FIXME: Remove when the native app switches over
+            'sections': [section_data(s) for s in sections],
+            'venues': [venue_data(venue) for venue in self.obj.venues],
+            'rooms': [room_data(room) for room in self.obj.rooms],
+            'proposals': [proposal_data(proposal) for proposal in proposals],
+            'schedule': schedule_data(self.obj),
+            })
+
+    @route('csv')
+    @render_with(json=True)
+    @requires_permission('view')
+    def csv(self):
+        if 'view-contactinfo' in self.obj.current_permissions:
+            usergroups = [ug.name for ug in self.obj.usergroups]
         else:
-            return redirect(project.url_for(), code=303)
-    else:
-        abort(400)
-
-
-@app.route('/<profile>/<project>/rsvp_list')
-@funnelapp.route('/<project>/rsvp_list', subdomain='<profile>')
-@lastuser.requires_login
-@load_models(
-    (Profile, {'name': 'profile'}, 'g.profile'),
-    ((Project, ProjectRedirect), {'name': 'project', 'profile': 'profile'}, 'project'),
-    permission='edit-project')
-def rsvp_list(profile, project):
-    return render_template('project_rsvp_list.html.jinja2', project=project, statuses=RSVP_STATUS)
-
-
-@app.route('/<profile>/<project>/transition', methods=['POST'])
-@funnelapp.route('/<project>/transition', methods=['POST', ], subdomain='<profile>')
-@lastuser.requires_login
-@load_models(
-    (Profile, {'name': 'profile'}, 'g.profile'),
-    ((Project, ProjectRedirect), {'name': 'project', 'profile': 'profile'}, 'project'),
-    permission='edit-project')
-def project_transition(profile, project):
-    transition_form = ProjectTransitionForm(obj=project)
-    if transition_form.validate_on_submit():  # check if the provided transition is valid
-        transition = getattr(project.current_access(),
-            transition_form.transition.data)
-        transition()  # call the transition
-        db.session.commit()
-        flash(transition.data['message'], 'success')
-    else:
-        flash(_("Invalid transition for this project."), 'error')
-        abort(403)
-    return redirect(project.url_for())
+            usergroups = []
+        proposals = Proposal.query.filter_by(project=self.obj).order_by(db.desc('created_at')).all()
+        outfile = StringIO()
+        out = unicodecsv.writer(outfile, encoding='utf-8')
+        out.writerow(proposal_headers + ['votes_' + group for group in usergroups] + ['status'])
+        for proposal in proposals:
+            out.writerow(proposal_data_flat(proposal, usergroups))
+        outfile.seek(0)
+        return Response(unicode(outfile.getvalue(), 'utf-8'), content_type='text/csv',
+            headers=[('Content-Disposition', 'attachment;filename="{project}.csv"'.format(project=self.obj.title))])
```

Comment:
We need to escape `self.obj.title`. If it has quotes (valid user input), it will break the header. We'll have to lookup valid characters for the `Content-Disposition` header since HTML escaping may not be appropriate here.

Reply:
Should we replace `title` with `name`?

Reply:
I checked [rfc6266](https://tools.ietf.org/html/rfc6266) and it mentions - 

> Avoid using non-ASCII characters in the filename parameter.
      Although most existing implementations will decode them as
      ISO-8859-1, some will apply heuristics to detect UTF-8, and thus
      might fail on certain names.

Reply:
So we can encode the title with ascii just to be safe.